### DISCUSSION TITLE
what is the base docker image that all other images use?

### DISCUSSION THREAD

**Discussion by wuast94** [blakeblackshear/frigate#14218] [Ask A Question]
To get changelogs shown with Renovate a docker container has to add the source label described in the OCI Image Format Specification.

For reference: https://github.com/renovatebot/renovate/blob/main/lib/modules/datasource/docker/readme.md

i would like to open a PR on this but i cant find the root of all docker containers. 
https://docs.docker.com/reference/dockerfile/#label
`Labels included in base or parent images (images in the FROM line) are inherited by your image. If a label already exists but with a different value, the most-recently-applied value overrides any previously-set value.`

so we only need to add it after the last FROM statement all images are based on. but that is the part i cant find sadly, when someone can point me there im happy to open a PR :)

**Comment by hawkeye217** [comment]
Frigate does not have a "root" of docker containers.

**Reply by wuast94** [reply]
so i would have to add it to the whole matrix of containers that gets build?

**Reply by NickM-27** [reply]
It depends what docker image you are looking at. The base image uses `deps` as the source and the target is called `frigate`

https://github.com/blakeblackshear/frigate/blob/0b71cfaf06d50eebffa15ef6bad197324a134c95/docker/main/Dockerfile#L310

The rockchip image pulls from the same `deps` source

https://github.com/blakeblackshear/frigate/blob/0b71cfaf06d50eebffa15ef6bad197324a134c95/docker/rockchip/Dockerfile#L12

the tensorrt image for example pulls in deps but uses tensorrt-base as the final source since the image needs to be based on the nvidia python image

https://github.com/blakeblackshear/frigate/blob/0b71cfaf06d50eebffa15ef6bad197324a134c95/docker/tensorrt/Dockerfile.amd64#L29

**Reply by NickM-27** [reply]
it's not clear what you are trying to accomplish or create a PR for

**Reply by wuast94** [reply]
i want to add this `LABEL org.opencontainers.image.source="https://github.com/blakeblackshear/frigate"` to all the containers, im just using the tensorrt one but this should be added to all images. 

this is from the OCI Image Format Specification mentioned above and gives the docker container metadata where the source lives that build this container. 

this metadata is used by tools like renovate to link that source or pull the changes written in the release. 

i personally use renovate to make updates to my swarm cluster, so renovates open a PR with all the information needed and if there is no breaking change tec i can just merge the pr and my cluster automaticlly updates the container to the newest version
